### PR TITLE
RDM-4810- To increase unit test coverage in "DefinitionStoreAPI" for CCD-AM Integration

### PR DESCRIPTION
**Pull request by vinoth-senthamaraikannan**
Jira Link:
https://tools.hmcts.net/jira/browse/RDM-4810

Change description
CaseTypeRepositoryTest - test methods have been added to improve the unit test coverage. The new test method helps to add CaseTypeACL information and retrieve the same.

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[X ] No
```

### CODE REVIEW

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java [hunk 1]
```diff
@@ -169,4 +215,15 @@ private void saveCaseTypeClearAndFlushSession(CaseTypeEntity caseType) {
...
     }
 
+    @Test
+    public void saveAndGetCaseTypeWithCaseTypeACLData() {
+        createCaseTypeEntityWithCaseTypeACL("CaseTypeWithACL","CaseTypeWithACL", 1,testJurisdictionWithCaseTypeACL,createCaseTypeACL());
+        List<CaseTypeEntity> caseTypeEntities
+            = classUnderTest.findByJurisdictionId(testJurisdictionWithCaseTypeACL.getReference());
+        assertEquals(1, caseTypeEntities.size());
+        assertEquals(4, caseTypeEntities.get(0).getCaseTypeACLEntities().size());
+
```
[the remark this answers is not in the document]
Comment (MSancaktutar) [reply repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java]: Please have a look at this comment.

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java [hunk 2]
```diff
@@ -169,4 +215,15 @@ private void saveCaseTypeClearAndFlushSession(CaseTypeEntity caseType) {
         entityManager.clear();
     }
 
+    @Test
+    public void saveAndGetCaseTypeWithCaseTypeACLData() {
+        createCaseTypeEntityWithCaseTypeACL("CaseTypeWithACL","CaseTypeWithACL", 1,testJurisdictionWithCaseTypeACL,createCaseTypeACL());
```
[the remark this answers is not in the document]
Comment (MSancaktutar) [reply repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java]: Please have a look at this comment.

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinoth-senthamaraikannan) [reply repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java]: Incorporated the comments. thank you

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (vinoth-senthamaraikannan) [reply repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java]: Incorporated the comments.

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java [hunk 3]
```diff
@@ -169,4 +224,54 @@ private void saveCaseTypeClearAndFlushSession(CaseTypeEntity caseType) {
         entityManager.clear();
     }
 
+    @Test
+    public void saveAndGetCaseTypeWithCaseTypeACLData() {
```
Comment (ngul) [inline repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepositoryTest.java]: Can we change this method name as discussed yesterday. This test does alot more and the name is not completely representative of that.

**Pull request merged**